service/history/replication/dlq_handler.go

```diff
@@ -120,9 +123,31 @@ func newDLQHandler(
 		resender: xdc.NewNDCHistoryResender(
 			shard.GetNamespaceRegistry(),
 			clientBean,
-			func(ctx context.Context, request *historyservice.ReplicateEventsV2Request) error {
-				_, err := clientBean.GetHistoryClient().ReplicateEventsV2(ctx, request)
-				return err
+			func(
```

Comment:
Could history resender DI from fx?

Reply:
Yes it could DI from fx. However, I believe the reason it's currently not DI from fx is that we seek flexibility across different scenarios.